mscore/instrwidget.cpp

```diff
@@ -478,6 +480,139 @@ void InstrumentsWidget::genPartList(Score* cs)
+//   sanitizePartiturListSelection
+//---------------------------------------------------------
+
+void InstrumentsWidget::sanitizePartiturListSelection()
+      {
+      _partiturSelectedItems.clear();
+            
+      if (partiturList->selectedItems().isEmpty() || partiturList->selectedItems().front() == nullptr) {
+            _partiturSelectionType = NONE_SELECTED;
+            return;
+            }
+            
+      // If a parent is selected, actions apply to all children as well. Select them, but keep track of parents.
+      // If children of multiple parents are selected, select all parents (and their children, by previous convention).
+      QSet<QTreeWidgetItem*> selectedParents;
+      bool containsParents = false;
+      for (QTreeWidgetItem* item : partiturList->selectedItems()) {
+            QTreeWidgetItem* parent = item->parent();
+            if (parent == NULL) {
+                  selectedParents.insert(item);
+                  containsParents = true;
+                  }
+            else {
+                  selectedParents.insert(parent);
+                  }
+            }
+      
+      _partiturSelectionType = (selectedParents.size() > 1 || containsParents) ? PARENTS_SELECTED : CHILDREN_SELECTED;
+      if (_partiturSelectionType == CHILDREN_SELECTED) {
+            for (QTreeWidgetItem* item : partiturList->selectedItems())
+                  _partiturSelectedItems.append(item);
+            }
+      else {
+            for (QTreeWidgetItem* parent : selectedParents) {
+                  _partiturSelectedItems.append(parent);
+                  parent->setSelected(true);
+                  for (int childId = 0; childId < parent->childCount(); childId++)
+                        parent->child(childId)->setSelected(true);
+                  
+                  }
+            }
+      }
+
+//---------------------------------------------------------
+//   movePartiturListItems
+//---------------------------------------------------------
+
+void InstrumentsWidget::movePartiturListItems(bool moveUp)
+      {
+            if (_partiturSelectionType == NONE_SELECTED) return;
+            
+            int delta = moveUp ? -1 : 1;
+            QList<QTreeWidgetItem*> originalSelection = partiturList->selectedItems();
+            QList<QTreeWidgetItem*> itemsToMove = QList<QTreeWidgetItem*>(_partiturSelectedItems);
+            PartiturSelectionType selectionType = _partiturSelectionType;
+            partiturList->selectionModel()->clear();
+            
+            QTreeWidgetItem* item;
+            if (selectionType == PARENTS_SELECTED) {
+                  int idx = delta > 0 ? partiturList->topLevelItemCount() - 1 : 0;
+                  for ( ; (item = partiturList->topLevelItem(idx)); idx += -delta) {
+                        if (!itemsToMove.contains(item))
+                              continue;
+                        
+                              bool isExpanded = partiturList->isItemExpanded(item);
```

Comment:
Seems the indentation issue starts here. This spot and on should match the indentation of the above if statement (subtract 6 spaces).